guides/dsl.md

```diff
@@ -0,0 +1,289 @@
+4) You can easily import data into Eve. If you can get your data in Javascript, you can use it in Eve.
+
+## Using the DSL
+
+To use the DSL, import it into your application.
+
+```javascript
+import {Program} from "./runtime/dsl2";
+```
+
+Then you instantiate a new Eve program, naming it however you want.
+
+```javascript
+let program = new Program("program name");
+```
+
+Writing DSL code is similar to writing native Eve code; you attach blocks to the `program`, which search for records and then return records. Blocks in the DSL operate as a either a `bind` or `commit`. Let's look at the DSL block syntax in the case of `bind`:
+
+```javascript
+  program.bind("block description", ({find, record}) => {
+    // search for records tagged "person"
+    let person = find("person");
+    return [
+      // bind a record tagged "greeting" with an attribute "message"
+      record("greeting", {message: `Hello ${person.name}`})
+    ];
+  });
+```
+
+The equivalent Eve block would be
+
+```eve
+search
+  person = [#person]
+
+bind
+  [#greeting message: "Hello {{person.name}}"]
+```
+
+The identity of a `record()` is established by everything within the parentheses. Therefore, the record `record("greeting", {message: `Hello ${person.name}`})` will create one record for every `person.name`.
+
+If would like to commit records instead of binding them, you can use the `commit()` method in place of `bind()`. The implication here is that you can no longer create blocks which both commit and bind records at the same time.
+
+The DSL supports methods which mirror a couple of the Eve update operators. They are:
+
+### add()
+
+ Adds attribute/value pairs to a record. These added values do not contribute to the identity of a record, allowing you to add multiple values to a single record. This is equivalent to the Eve `+=` operator. For example:
+
+```javascript
+program.bind("Invite classmates to my party", ({find, record}) => {
+  // search for records tagged "friend"
+  let student = find("student");
+  return [
+    // bind a single record tagged "guest-list", with every student as a guest
+    record("guest-list").add({guest: student})
+  ];
+});
+```
+
+### remove()
+
+Removes attribute/value pairs from a record. This is equivalent to the Eve `-=` operator. Remove should only be used in the context of a `commit` block. For example:
+
+```javascript
+program.commit("blacklisted people cannot come to the party", ({find, record}) => {
+  let blacklisted = find("blacklisted");
+  let guests = find("guest-list")
+  return [
+    // remove anyone blacklisted from the guest list
+    guests.remove({guest: blacklisted})
+  ];
+});
+```
+
+You can also use `remove()` to completely delete records or attributes on a record. This is equivalent to `:= none` in Eve. For example:
+
+```javascript
+   // completely remove the target record
+   record.remove() 
+   // remove the attribute on the target record
+   record.remove("attribute")
```

Comment:
Let's use an attribute other than "attribute" here to make sure it's obvious that this is an example value and not a magic keyword.